src/apis/ApiHandlers.ts

```diff
@@ -24,110 +46,108 @@ export class ApiHandler {
     //--------------------------------------------------------------------------------------
     // 
     //--------------------------------------------------------------------------------------
-    serverError(res: Response, err: any, errorType: string) {
-        this.serverModel.reportError(errorType)
-        const key = Date.now();
-        this.logger.logError(`API Error: key: ${key} :${err as any}`);
-        res.status(400).end(`Server error.  Key = ${key} `);
+    async safeCall(req: Request, res: Response, label: string, runMe: () => Promise<any>) {
+        try {
+            const data = await runMe();
+            res.setHeader('Content-Type', 'application/json');
+            res.end(JSON.stringify(data, null, 2));
+        }
+        catch(err) {
+            if(err instanceof UserError){
+                const errorResponse = {
+                    errorMessage: (err as UserError).message   
+                }
+                res.status(400).end(JSON.stringify(errorResponse, null, 2))
+            }
+            else {
+                const timecode = Date.now();
+                this.logger.logError(`Error at timecode ${timecode} on ${label}: ${err} ${JSON.stringify(err)}`)
+                const errorResponse = {
+                    errorMessage: `There was a server error in ${label}.  Reference timecode ${timecode}`   
+                }
+                res.status(400).end(JSON.stringify(errorResponse, null, 2))
+            }
+        }
```

Comment:
Nit: I'd suggest a currying model here - rather than have to call `this.safeCall()` in every API request, we can fill our API methods directly with safely wrapped calls:

    async makeSafeCall(label: string, runMe: async (req: Request) => Promise<any>) {
        return (req: Request, res: Response) => {
            /* same logic as before - run runMe with the request provided and return the data you get back */
        }
    }

    showHealth = makeSafeCall("ShowHealth, async (req: Request) => {
        let span = req.query.span ? Number.parseInt(req.query.span as string) : 60000
        // etc.
     })